discovery-provider/integration_tests/tasks/test_anchor_program_indexer.py

```diff
@@ -111,6 +121,49 @@ def test_parse_tx(app):
     assert owner_eth_address_hex == "0x25A3Acd4758Ab107ea0Bd739382B8130cD1F204d"
 
 
+def test_fetch_metadata(app):
+    with app.app_context():
+        db = get_db()
+        redis = get_redis()
+
+    populate_mock_db(db, basic_entities, block_offset=3)
+
+    solana_client_manager_mock = create_autospec(SolanaClientManager)
+    cid_metadata_client_mock = create_autospec(CIDMetadataClient)
+
+    cid_metadata_client_mock.fetch_metadata_from_gateway_endpoints.return_value = (
+        mock_cid_metadata
+    )
+
+    anchor_program_indexer = AnchorProgramIndexer(
+        PROGRAM_ID,
+        ADMIN_STORAGE_PUBLIC_KEY,
+        LABEL,
+        redis,
+        db,
+        solana_client_manager_mock,
+        cid_metadata_client_mock,
+    )
+    parsed_tx = {
+        "tx_metadata": {
+            "instructions": [
+                {
+                    "instruction_name": "init_user",
+                    "data": Container(
+                        [("metadata", list(mock_cid_metadata.keys())[0])]
+                    ),
+                }
+            ]
+        },
+        "tx_sig": "x4PCuQs3ncvhJ3Qz18CBzYg26KnG1tAD1QvZG9B6oBZbR8cJrat2MzcvCbjtMMn9Mkc4C8w23LHTFaLG4dJaXkV",
+    }
+    mock_parsed_transactions = [parsed_tx]
+    cid_metadata = asyncio.run(
+        anchor_program_indexer.fetch_ipfs_metadata(mock_parsed_transactions)
```

Comment:
running into an issue with nested async runs. one here and one in the client. the regular indexer isn't async so it doesn't run into this.